jazz_services-handler/index.js

```diff
@@ -20,584 +21,277 @@ module.exports.handler = (event, context, cb) => {
-																	"error": err.error
-																});
-															} else {
-																HandleProcessedEvents(sequenceNumber, payload);
-																logger.info("updated service " + payload.SERVICE_NAME.S + " in service catalog.");
-																return innerCallback(null, {
-																	"message": "updated service " + payload.SERVICE_NAME.S + " in service catalog."
-																});
-															}
-														});
-													}
-												});
-
-											}
-
-										}
-
-									}
-									else if (payload.EVENT_TYPE.S === "SERVICE_DEPLOYMENT") {
-
-										if (payload.EVENT_NAME.S === configData.SERVICE_DEPLOYMENT_EVENT_END && payload.EVENT_STATUS.S === "COMPLETED") {
-											//delete service in catalog with status deletion_started
-											inputs = {
-												"TOKEN": authToken,
-												"SERVICE_API_URL": configData.SERVICE_API_URL,
-												"SERVICE_API_RESOURCE": configData.SERVICE_API_RESOURCE,
-												"DOMAIN": domain,
-												"SERVICE_NAME": payload.SERVICE_NAME.S
-											};
-											crud.get(inputs, function (err, results) {
-												if (err) {
-													HandleFailedEvents(sequenceNumber, err.error, payload, failureCodes.PR_ERROR_4.code);
-													logger.error(err.details);
-													return innerCallback({
-														"error": err.error
-													});
-												} else {
-
-													inputs = {
-														"TOKEN": authToken,
-														"SERVICE_API_URL": configData.SERVICE_API_URL,
-														"SERVICE_API_RESOURCE": configData.SERVICE_API_RESOURCE,
-														"ID": results.id,
-														"DESCRIPTION": description,
-														"REPOSITORY": repository,
-														"EMAIL": email,
-														"SLACKCHANNEL": slackChannel,
-														"TAGS": tags,
-														"ENDPOINTS": endpoint,
-														"STATUS": "active"
-													};
-
-													crud.update(inputs, function (err, results) {
-														if (err) {
-															HandleFailedEvents(sequenceNumber, err.error, payload, failureCodes.PR_ERROR_2.code);
-															logger.error(err.details);
-															return innerCallback({
-																"error": "unknown error updating service " + payload.SERVICE_NAME.S + " in service catalog."
-															});
-														} else {
-															HandleProcessedEvents(sequenceNumber, payload);
-															logger.info("updated service "  + payload.SERVICE_NAME.S + " in service catalog.");
-															return innerCallback(null, {
-																"message": "updated service "  + payload.SERVICE_NAME.S + " in service catalog."
-															});
-														}
-													});
-
-												}
-
-											});
-										}
-									}
+		}; //End of promise			
+	}
+	
+	function validateAuthToken(result){
+		return new Promise((resolve, reject) => {
+			if (result.statusCode === 200 && typeof result.body !== undefined && typeof result.body.data !== undefined) {
```

Comment:
result.statusCode === 200 && result.body && result.body.data

Reply:
In cooperated the comment